guides/Diagnostics.md

```diff
@@ -0,0 +1,62 @@
+# Diagnostics
+In this doc we are going to introduce you to our application diagnostics and logging style and approach.
+
+## The Challenge
+
+We use the corporate [papertrail](https://papertrailapp.com) account for hosted log management. Usually we only use it for production environments. The main problem we face here is how to reduce logs size and avoid noisy and unusable logs.
+
+## The Solution
+
+[lograge](https://github.com/roidrage/lograge) is a gem which helps us with application logging configuration. You can find the recommended gem configuration below:
+
+```ruby
+# config/initializers/lograge.rb
+
+Rails.application.configure do
+  config.lograge.enabled = Rails.env.production?
+  config.lograge.custom_options = lambda do |event|
+    exceptions = %w[controller action format]
+    {
+      params: event.payload[:params].except(*exceptions)
+    }
+  end
+end
+```
+
+If you’re using Rails 5's API-only mode and inherit from `ActionController::API`, you must define it as the controller base class which lograge will patch:
+
+```ruby
+config.lograge.base_controller_class = 'ActionController::API'
+```
+
+### Application configuration
+
+Use `info` log level to remove unnecessary debugging logs.
+```ruby
+# config/environments/production.rb
+
+config.log_level = :info
+```
+
+## Pepertail setting up
+
+Creating a new papertrail system for Heroku applications - [https://papertrailapp.com/systems/setup?type=app&platform=heroku]()
+
+# Error reporting API
+
+As with the [papertrail](https://papertrailapp.com) corporate account we have [Sentry](https://sentry.io) account. The installation guide you can find by [this link](https://github.com/getsentry/raven-ruby). The only one thing what you have to do after gem installation is to set `SENTRY_DSN` ENV variable. The value of that variable you can find on [new project creation page](https://sentry.io/organizations/dvelp/projects/new).
```

Comment:
I think you should put the Sentry installation and config in this doc. The idea is to make it easy for people to reference without having to head off to different pages.

Depending on where you look, you will get different instructions and the aim of this document is to insure consistency of approach across all codebases.